### ISSUE TITLE
[BUG] YouTube throttling videos

### ISSUE THREAD

**Issue by itssidhere** [Hexer10/youtube_explode_dart#180]
Original issue is [here](https://github.com/Tyrrrz/YoutubeExplode/issues/599#issue-1035765454). We should keep one instance of this issue open here also. Thanks

**Comment by itssidhere** [comment]
Please have a look at this [pull request](https://github.com/Tyrrrz/YoutubeExplode/pull/601) I am facing severe throttling.

**Comment by lrorpilla** [comment]
Not sure if it's known already but my app which I haven't updated and am using an older version of the package was working fine. On the latest `master`, the throttling issue is pretty severe and takes a while to load up videos but `a51e38dcf8299f11271511922dc90ac48e7f5a47` is working perfectly for me.

I'm not sure or informed enough to make the connection, but anyways, reverting from `master` to this got me up and running again with no problems or throttling.

**Comment by itssidhere** [comment]
@lrorpilla Can you share the exact code from your dependency please?

**Comment by lrorpilla** [comment]
I'm not sure I understand -- I just noticed severe throttling on `master` and after making the revert, videos stopped halting every 5 seconds or so. The only thing I do is search, get the manifest and then plug the stream URLs to `flutter_vlc_player` -- I didn't touch `flutter_vlc_player` to make any changes.

I didn't make any changes to the package's source code, I just switched the ref.

**Comment by itssidhere** [comment]
@lrorpilla I do the exact same. Can you just tell me which version you are using of this package?

**Comment by lrorpilla** [comment]
I use this `https://github.com/Hexer10/youtube_explode_dart/tree/a51e38dcf8299f11271511922dc90ac48e7f5a47/`, and this is the dependency.
```
youtube_explode_dart:
    git:
      url: https://github.com/Hexer10/youtube_explode_dart
      ref: a51e38dcf8299f11271511922dc90ac48e7f5a47
```

**Comment by itssidhere** [comment]
Ok thanks, I will try this too

**Comment by itssidhere** [comment]
@lrorpilla Damn It fixed you are a hero! Now only thing we need is what broke it so we can fix it in latest version

**Comment by itssidhere** [comment]
@Hexer10 Original issue is closed now. Can you check once we can work on fixing for this plugin also!

**Comment by Hexer10** [comment]
I'm starting having a look on how it was fixed.
The main problem is that some time ago some commits changed the directory structure and some other stuff, so that may need to be ported as well.

Currently I also have some exams due to next month, so I'm not sure when I'll be able to push a fix

**Comment by itssidhere** [comment]
Also just to mention the solution mentioned by @lrorpilla doesn't work for some videos example - https://youtu.be/s1-D8UOq7iA

**Comment by itssidhere** [comment]
@Hexer10 I have managed to fix the issue on the latest 1.10.8 . Please review it
